### PR TITLE
implemented waist-circumference api using lumen

### PR DESCRIPTION
**Pull request by shubhadeep-grmtech**

### CODE REVIEW

**Review by rajgrmtech — APPROVED** [review]

File: servers/lumen/app/Http/Controllers/WaistCircumferenceController.php [hunk 1]
```diff
@@ -6,45 +6,76 @@
...
-    public function getOneWaistCircumference($pServerSideRowUuid)
+    public function getOneWaistCircumference($serverSideRowUuid)
     {
-        return response()->json(WaistCircumference::find($pServerSideRowUuid));
+        $waistCircumferenceQuery = DB::select(DB::raw("SELECT *, round(UNIX_TIMESTAMP(ROW_START) * 1000) as ROW_START, round(UNIX_TIMESTAMP(ROW_END) * 1000) as ROW_END, UNIX_TIMESTAMP(timeOfMeasurementInMilliseconds) * 1000 as timeOfMeasurementInMilliseconds FROM sc_vital_signs.waistCircumference FOR SYSTEM_TIME ALL WHERE serverSideRowUuid LIKE '{$serverSideRowUuid}' order by ROW_START desc"));
+
+        return response()->json($waistCircumferenceQuery);
     }
 
     public function create(Request $request)
```
Comment (vikaskedia) [inline servers/lumen/app/Http/Controllers/WaistCircumferenceController.php:29]: This should be $pRequest

File: servers/lumen/app/Http/Controllers/WaistCircumferenceController.php [hunk 2]
```diff
@@ -6,45 +6,76 @@
...
 
-        $waistCircumferenceObj = WaistCircumference::insertGetId($waistCircumferenceData);
+        $insertWaistCircumference = DB::statement("INSERT INTO `sc_vital_signs`.`waistCircumference` (`serverSideRowUuid`, `ptUuid`, `waistCircumferenceInInches`, `timeOfMeasurementInMilliseconds`, `notes`, `recordChangedByUuid`, `recordChangedFromIPAddress`) VALUES ('{$serverSideRowUuid}', '{$ptUuid}', {$waistCircumferenceInInches}, FROM_UNIXTIME({$timeOfMeasurementInMilliseconds}/1000), '{$notes}', '{$recordChangedByUuid}', '{$recordChangedFromIPAddress}')");
 
-        return response()->json($waistCircumferenceObj, 201);
+        return response()->json($insertWaistCircumference, 201);
     }
 
     public function update($serverSideRowUuid, Request $request)
     {
```
Comment (vikaskedia) [inline servers/lumen/app/Http/Controllers/WaistCircumferenceController.php:47]: This should be pRequest also

File: servers/lumen/app/Http/Controllers/WaistCircumferenceController.php [hunk 3]
```diff
@@ -6,45 +6,76 @@
...
 
-        $waistCircumferenceData = array(
-            'serverSideRowUuid' => $requestData['data']['serverSideRowUuid'],
-            'ptUuid' => $requestData['data']['ptUuid'],
-            'waistCircumferenceInInches' => $requestData['data']['waistCircumferenceInInches'],
-            'timeOfMeasurement' => $requestData['data']['timeOfMeasurement'],
-            'notes' => $requestData['data']['notes'],
-            'recordChangedByUuid' => $requestData['data']['recordChangedByUuid']
-        );
+        $serverSideRowUuid = $requestData['data']['serverSideRowUuid'];
```
Comment (vikaskedia) [inline servers/lumen/app/Http/Controllers/WaistCircumferenceController.php:33]: this needs to be pServerSideRowUuid

File: webclient/components/ptinfo-single/1time-1row-mField/vital-signs/sub-cts/waist-circumference/db/server-side/structure.sql [hunk 4]
```diff
@@ -6,7 +6,7 @@ CREATE TABLE `waistCircumference` (
   `serverSideRowUuid` char(36) CHARACTER SET utf8 COLLATE utf8_unicode_ci NOT NULL,
   `ptUuid` char(36) CHARACTER SET utf8 COLLATE utf8_unicode_ci DEFAULT NULL,
   `waistCircumferenceInInches` int(11) DEFAULT NULL,
-  `timeOfMeasurementInMilliseconds` datetime DEFAULT NULL,
+  `timeOfMeasurementInMilliseconds` datetime(3) NOT NULL DEFAULT current_timestamp(3),
```
Comment (vikaskedia) [inline webclient/components/ptinfo-single/1time-1row-mField/vital-signs/sub-cts/waist-circumference/db/server-side/structure.sql:9]: give link to tech/how.md Q13 (check for exact Q number)

**Review by vikaskedia — CHANGES_REQUESTED** [review]

File: servers/lumen/app/Http/Controllers/WaistCircumferenceController.php [hunk 5]
```diff
@@ -6,45 +6,76 @@
...
+
 \Predis\Autoloader::register();
 
+
 class WaistCircumferenceController extends Controller
 {
     public function getAllTemporalWaistCircumferences()
     {
-        $waistCircumferenceQuery = DB::select(DB::raw('SELECT *, round(UNIX_TIMESTAMP(ROW_START) * 1000) as ROW_START, round(UNIX_TIMESTAMP(ROW_END) * 1000) as ROW_END FROM sc_vital_signs.waistCircumference FOR SYSTEM_TIME ALL order by ROW_START desc'));
+        $waistCircumferenceQuery = DB::select(DB::raw('SELECT *, round(UNIX_TIMESTAMP(ROW_START) * 1000) as ROW_START, round(UNIX_TIMESTAMP(ROW_END) * 1000) as ROW_END, UNIX_TIMESTAMP(timeOfMeasurementInMilliseconds) * 1000 as timeOfMeasurementInMilliseconds FROM sc_vital_signs.waistCircumference FOR SYSTEM_TIME ALL order by ROW_START desc'));
```
Comment (vikaskedia) [inline servers/lumen/app/Http/Controllers/WaistCircumferenceController.php:17]: explain the reason why you are doing the round.

Comment (vikaskedia) [reply servers/lumen/app/Http/Controllers/WaistCircumferenceController.php:17]: explain the reason as a comment inside the code

File: servers/lumen/app/Http/Controllers/WaistCircumferenceController.php [hunk 6]
```diff
@@ -6,45 +6,76 @@
...
 use DB;
 use Predis\Autoloader;
+
 \Predis\Autoloader::register();
 
+
 class WaistCircumferenceController extends Controller
 {
     public function getAllTemporalWaistCircumferences()
     {
```
Comment (vikaskedia) [inline servers/lumen/app/Http/Controllers/WaistCircumferenceController.php:16]: Do you think the name waistCircumferenceQueryResultObj is better?

When I read waistCircumferenceQuery i think it is just the query

File: servers/lumen/app/Http/Controllers/WaistCircumferenceController.php [hunk 7]
```diff
@@ -6,45 +6,76 @@
...
+        $waistCircumferenceQuery = DB::select(DB::raw('SELECT *, round(UNIX_TIMESTAMP(ROW_START) * 1000) as ROW_START, round(UNIX_TIMESTAMP(ROW_END) * 1000) as ROW_END, UNIX_TIMESTAMP(timeOfMeasurementInMilliseconds) * 1000 as timeOfMeasurementInMilliseconds FROM sc_vital_signs.waistCircumference FOR SYSTEM_TIME ALL order by ROW_START desc'));
+
         return response()->json($waistCircumferenceQuery);
     }
 
-    public function getOneWaistCircumference($pServerSideRowUuid)
+    public function getOneWaistCircumference($serverSideRowUuid)
     {
-        return response()->json(WaistCircumference::find($pServerSideRowUuid));
+        $waistCircumferenceQuery = DB::select(DB::raw("SELECT *, round(UNIX_TIMESTAMP(ROW_START) * 1000) as ROW_START, round(UNIX_TIMESTAMP(ROW_END) * 1000) as ROW_END, UNIX_TIMESTAMP(timeOfMeasurementInMilliseconds) * 1000 as timeOfMeasurementInMilliseconds FROM sc_vital_signs.waistCircumference FOR SYSTEM_TIME ALL WHERE serverSideRowUuid LIKE '{$serverSideRowUuid}' order by ROW_START desc"));
```
Comment (vikaskedia) [inline servers/lumen/app/Http/Controllers/WaistCircumferenceController.php]: some times you have rounded and others you have not?

File: servers/lumen/app/Http/Controllers/WaistCircumferenceController.php [hunk 8]
```diff
@@ -6,45 +6,76 @@
...
+
+        return response()->json($waistCircumferenceQuery);
```
Comment (vikaskedia) [inline servers/lumen/app/Http/Controllers/WaistCircumferenceController.php:26]: improve variable name

**Pull request closed without merging**